traitsui/extras/demo.py

```diff
@@ -121,7 +119,7 @@ def parse_source(file_name):
         The source code, sans docstring.
     """
     try:
-        with src_open(file_name, 'r') as fh:
+        with open(file_name, 'r') as fh:
```

Comment:
In general, you should always include an explicit encoding with `io.open`. If you don't, it uses the "default encoding", which is a somewhat murky thing and may vary not just from OS to OS but from user to user.

I'm not sure of the context here, but either "ascii" or "utf-8" would probably be suitable encodings to use.

And a nitpick: I have a mild preference for `import io` and using `io.open`, rather than `from io import open` and `open(...)`. That gives the local context so that readers of the code are aware that this _isn't_ the regular built-in open.

Reply:
@mdickinson Thanks for the review and the sensible recommendations. I push fixes.

I didn't specify an encoding because the previous code used `tokenize.open`, which itself relies on `tokenize.detect_encoding` and… I didn't read the docs until the end where it says it uses `utf-8` by default if it can't figure out anything better.

Reply:
We control all the source code it should be reading (these are demo source files) so we can assume any encoding we like, as long as we are consistent with the encoding of the actual files.  In other words, we can safely assume `utf-8` if it makes life easier.